containers.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2034,SC2143
+set -euo pipefail
+
+# Network array - to add more, separate by space and double qoutes e.g. "todo-app" "your-second-network"
+networks=("todo-app")
+# Volume array - to add more, separate by space and double qoutes, e.g. "todo-mysql-data" "your-second-volume"
+volumes=("todo-mysql-data")
+# Path to docker images
+images="${PWD}/docker/images"
+# Set app container name
+app="todo-app"
+# Set db container name
+db="todo-db"
+
+log(){
+  echo "$(date +'%T') $1"
+}
+
+networks(){
+  docker network create \
+    "${network}"
+}
+
+volumes(){
+  docker volume create \
+    "${volume}"
+}
+
+todo_app(){
+  docker run \
+    --detach \
+    --name "${app}" \
+    --publish 3000:3000 \
+    --workdir "/app" \
+    --volume "${images}/getting-started/app:/app" \
+    --network todo-app \
+    --env MYSQL_HOST=mysql \
+    --env MYSQL_USER=root \
+    --env MYSQL_PASSWORD=secret \
+    --env MYSQL_DB=todos \
+    node:18-alpine \
+    sh -c "yarn install && yarn run dev"
+}
+
+todo_db(){
+  docker run \
+    --detach \
+    --name "${db}" \
+    --network todo-app \
+    --network-alias mysql \
+    --volume todo-mysql-data:/var/lib/mysql \
+    --env MYSQL_ROOT_PASSWORD=secret \
+    --env MYSQL_DATABASE=todos \
+     mysql:5.7
+}
+
+# Create networks for containers
```

Comment:
Ok. I intentionally sourced it to re-use my variables and arrays in gc.sh, but `containers.sh`is designed to run locally (spawn containers and avoid terminating them in the same script)
I can remove the `.sh` bit from gc.sh if you think that's the best practice.

Reply:
Nah, just consider it the next time - as you write larger and larger scripts you start to rely on some of these conventions to avoid running in to bugs. Eg, having some sourced file call a function on inclusion because it did not just contain functions but also some top-level code.